bip-0078.mediawiki

```diff
@@ -106,14 +106,20 @@ The original PSBT MUST:
 The original PSBT MAY:
 * Have outputs unrelated to the payment for batching purpose.
 
+The original PSBT SHOULD NOT:
+* Include mixed input types until September 2024. Mixed inputs were previously completely disallowed so this gives some grace period for receivers to update.
+
 The payjoin proposal MUST:
 * Use all the inputs from the original PSBT.
 * Use all the outputs which do not belong to the receiver from the original PSBT.
 * Only finalize the inputs added by the receiver. (Referred later as <code>additional inputs</code>)
 * Only fill the <code>witnessUTXO</code> or <code>nonWitnessUTXO</code> for the additional inputs.
 
 The payjoin proposal MAY:
-* Add, remove or modify the outputs belonging to the receiver.
+* Add, or replace the outputs belonging to the receiver unless output substitution is disabled.
+
+The payjoin proposal SHOULD NOT:
+* Include mixed input types until September 2024. Mixed inputs were previously completely disallowed so this gives some grace period for senders to update.
```

Comment:
@DanGould As a follow-up, do you think these dates here and on line 110 above ought to be updated to provide a post-merge grace period?

Reply:
A client may still opt to disallow mixed inputs (or any payjoin proposal) by policy rather than by protocol like this one, so I don't think it's so critical that requires an explicit grace period after all, nor does the change in #1396. I think we can leave the grace period as-is and tackle the implementations one by one.

Reply:
Thanks! I also see that you are reaching out to implementations at the moment 👍